### DISCUSSION TITLE
dynamic dispatch ?

### DISCUSSION THREAD

**Discussion by rraammiinn** [vlang/v#17498] [Questions and Answers]
does it have any equivalent for dyn keyword in rust ?

**Comment by JalonSolov** [comment]
`dyn` is a trait in Rust.  V doesn't have traits.

There is nothing like `dyn` in V.

**Comment by Wajinn** [comment]
> does it have any equivalent for dyn keyword in rust ?

I'm confused by what is the purpose.  What V has and can do, is primarily shown in its [documentation](https://github.com/vlang/v/blob/master/doc/docs.md) and [examples](https://github.com/vlang/V/tree/master/examples).  Programming languages are different from each in terms of design, purpose, tools, community, etc...  So we can't expect one programming language to be and do exactly the same as the other.  There are going to be differences.  And that's OK, different tools, do different things.

Simply throwing out keywords from other languages is problematic, because it's asking people to know that other language.  Maybe people don't want to study that other language or do somebody else's homework.  It's like me going on a C++ forum, start referring or talking about Pascal, then expect others to know Pascal too.  Perhaps if you gave example code or clearly stated what you are trying to do, then people could be more helpful or offer better guidance.

**Reply by rraammiinn** [reply]
sorry for any inconvenience .
what I'm trying to say is that even rust with its  obsession about safety and performance has support for dynamic typing into some extend .
I think it's ok for vlang to be a little more flexible in typing .

**Reply by bogen85** [reply]
The `dynamic` in `dynamic dispatch` has nothing to do with the `dynamic` in `dynamic typing`. (Apart from both occurring at run time).

**Reply by Wajinn** [reply]
> I think it's ok for vlang to be a little more flexible in typing .

But that seems like a preference, which is going against basic V design decisions, not so much an answerable question.  As you might prefer one thing, others may prefer something else.  Maybe a better way to present that argument is to show how such a feature would benefit the V community.  You could write it out in pseudo code, and make the argument that if V had this feature, we could do X.

Just stating that Y language has a certain feature, doesn't mean it's a benefit to X language.  So might be better to make an argument demonstrating the benefit.

**Reply by bogen85** [reply]
>  Maybe people don't want to study that other language or do somebody else's homework.

@Wajinn makes a good point.

There are also some who don't do the homework themselves and they expect others to do the homework for them.

**Comment by bogen85** [comment]
What does https://en.wikipedia.org/wiki/Dynamic_dispatch have to do with https://en.wikipedia.org/wiki/Type_system#DYNAMIC ?

Dynamic dispatch is (at a very low level) about changing function pointers at runtime to get different behavior for some method.

**Comment by rraammiinn** [comment]
sorry I was wrong .
it's functionality can be achieved using interfaces .

**Comment by rraammiinn** [comment]
can I also use empty interface just like in go ?

**Reply by bogen85** [reply]
https://github.com/vlang/v/search?q=%22empty+interface%22

https://github.com/vlang/v/issues/14162
https://github.com/vlang/v/issues/14244

**Reply by rraammiinn** [reply]
why not have a keyword for any when something like this  ( #14162 ) is possible ?
it would be just a syntax sugar so why not have it to get rid of declaring empty interface ?
